react-search/my-app/src/SearchableList.tsx

```diff
@@ -0,0 +1,28 @@
+import { useState } from 'react';
+import { QuotesList } from './QuotesList.tsx';
+import { SearchBar } from './SearchBar.tsx';
+
+type Props = {
+  quotes: string[];
+};
+export function SearchableList({ quotes }: Props) {
+  const [searchInput, setSearchInput] = useState('');
+
+  const filteredQuotes: string[] = [];
+  quotes.forEach((quote) => {
+    if (quote.toLowerCase().includes(searchInput.toLowerCase())) {
+      filteredQuotes.push(quote);
+    }
+  });
+
+  return (
+    <div>
+      <SearchBar input={searchInput} onInputChange={setSearchInput} />
+      {filteredQuotes.length ? (
+        <QuotesList quotes={filteredQuotes} />
+      ) : (
+        <p>No items match the filter.</p>
```

Comment:
I would probably have put this logic in QuotesList, but there are good arguments for both implementations. I just like having the QuotesList fully self-contained so I don't have to worry about the "no matches" part when I'm using it.